### PR TITLE
(384)Translation-Of-Home-And-Landing-Page

### LINKED ISSUE #384: CMS: Translation of Home and Landing page
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
  Scenario: Add a translation
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        Then user selects "Translate" option from Operations dropdown for content with title "Automated Test Home and Landing Page Edited"
        Then the page title is "Translations of Automated Test Home and Landing Page Edited"
        When user clicks on "Add" button to add translation
        Then page title is "Crear traducción Español de Automated Test Home and Landing Page Edited"
        And the following fields are displayed
            | fieldLabel               | field_name                     |
            | Pretty URL               | field_pretty_url               |
            | Título de página         | title                          |
            | Browser Title            | field_browser_title            |
            | Card Title               | field_card_title               |
            | Meta Description         | field_page_description         |
            | Feature Card Description | field_feature_card_description |
        And Hero Banner displays text "Aún no se ha añadido ningún Paragraph."
        And button to add a banner was translated as "Añadir Hero Banner"
        And the following sections have title field translated as "Título"
            | section                  |
            | Primary Feature Card     |
            | Guide Card Row           |
            | Listado                  |
            | One Column Container     |
            | Primary Feature Card Row |
        And user remembers the name of media card item for future verification
        And user fills out the following fields
            | fieldLabel        | value    | field_name                                                                                       |
            | Featured Item Url | spanish  | field_landing_contents[0][subform][field_row_cards][1][subform][field_featured_url][0][uri]      |
            | Title             | _Spanish | field_landing_contents[4][subform][field_list_title][0][value]                                   |
            | Override Title    | _Spanish | field_landing_contents[4][subform][field_list_items][2][subform][field_override_title][0][value] |
            | Button Text       | _Spanish | field_landing_contents[7][subform][field_button_text][0][value]                                  |
        And user selects "Archive Block" from Display dropdown
        And user clicks "Opciones" link in the Dynamic List area
        And user checks Include View Title checkbox
        And user enters "6" into Items per page dropdown
        And user selects "Minipaginador" from Pagination dropdown
        And current state was translated as "Estado actual" "Borrador"
        And Change to dropdown has the following options
            | option    |
            | Borrador  |
            | Review    |
            | Publicado |
        And Save button was translated as "Guardar (esta traducción)"
        And preview button was translated as "Vista previa"
        Then user saves the content page
        And user clicks on the tool bar status green button "Borrador"
        And user selects "Quick Publish" from workflow actions

    Scenario: Verify translated content
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "home-and-landing-page-edited"
        Then page title is "Automated Test Home and Landing Page Edited"
        And primary feature card row displays the following cards
            | title                             | link                          |
            | Article to test Related Resources | {TEST_SITE_SECTION}/article   |
            | Featured Item Url                 | https://www.google.comspanish |
        And list card row title is "List Title_Spanish"
        And list row displays the following links
            | title                                        | link                          | description                                          |
            | Article to test Related Resources            | {TEST_SITE_SECTION}/article   | Article to test Related Resources - Meta Description |
            | Google Link List Item                        | https://www.google2.com       |                                                      |
            | Media Link Override Title List Items_Spanish | {TEST_SITE_SECTION}/test-file |                                                      |
        And guide card row title is "Guide Card Row Title"
        And 2 guide card has text "This is to test Raw Html Content under Guide Card Row"
        And multimedia card row has a video which name matches selected multimedia card item
        And list card row title is "List Title"
        And one-column card row title is "One Column Container Title"
        And one-column content displays text "This is Html content for Content Block in One Column Container"
        And one-column list has raw html displaying "This is Html content for Content Block in One Column Container"
        And two-column list has raw html displaying "This is to test Raw HTML Content in Two Columns container"
        And two-column content heading reads "Content Heading Two Column Container"
        And two-column html content reads "This is Html content for Content Block Two Column"
        And two-column dynamic list shows "Press Releases"
        And dynamic lists shows 6 items
        And every link in dynamic list starts with "/espanol"
        And borderless card title is "Override title for Borderless Card"
        And borderless card long description is "Longer Override Card Description for Borderless Card"
        And borderless card button has title text "BUTTON TEXT FOR BORDERLESS FULL-WIDTH CARD_Spanish" and links to "{TEST_SITE_SECTION}/article"
        And the page contains meta tags with the following names
            | name        | content                                               |
            | description | Automated Test Home and Landing page Meta Description |

    Scenario: Edit and republish Spansih article content type
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url spanish path "/espanol" site section plus "home-and-landing-page-edited"
        And user clicks on the tool bar status green button "Publicado"
        And user clicks "View in edit form" button from other actions
        And user fills out the following fields
            | fieldLabel       | value    | field_name             |
            | Título de página | _Spanish | title                  |
            | Meta Description | _Spanish | field_page_description |
        When user saves the content page
        And user clicks on the tool bar status green button "Editing"
        And user selects "Quick Publish" from workflow actions

    Scenario: Verify Spanish edited content
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "home-and-landing-page-edited"
        And browser waits
        Then page title is "Automated Test Home and Landing Page Edited_Spanish"
        And the page contains meta tags with the following names
            | name        | content                                                         |
            | description | Automated Test Home and Landing Meta Description Edited_Spanish |

    Scenario: Clean up
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user selects a checkbox next to title with url "home-and-landing-page-edited" from the list of content
        And user clicks on "Apply to selected items" content action button
        Then page title is "Are you sure you want to delete this content item?"
        When user clicks on "Delete" button
        Then the confirmation text "Deleted 2 content items" appears on a screen
        And the content item with url "home-and-landing-page-edited" does not exist in the list of content
```
### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by ahmadakbaryNIH**
closes #384

### CODE REVIEW

**Comment by ahmadakbaryNIH** [comment]
image

File: cypress/e2e/CMS/$HomeAndLandingPage/index.js [hunk 1]
```diff
@@ -469,4 +469,41 @@ Then('the promo image is matching the earlier selected image', () => {
...
+And('Hero Banner displays text {string}', (TextDisplay) => {
+    cy.get(`div[id*='hero-banner-text'] em:contains("${TextDisplay}")`).should('be.visible')
+})
+
+And('button to add a banner was translated as {string}', (textTranslated) => {
+    cy.get(`input[value^="${textTranslated}"]`).should('be.visible')
+})
+
+And('the following sections have title field translated as {string}', (spPath, dataTable) => {
+    cy.get(`div[class*='contents-9-subform-field-'] label:contains("${spPath}")`).should('be.visible')
```
[the remark this answers is not in the document]
Comment (KateMashkinaNIH) [reply cypress/e2e/CMS/$HomeAndLandingPage/index.js:483]: you should actually find a section provided from data table and search for the Titulo element within the scope of that section

File: cypress/e2e/CMS/$HomeAndLandingPage/index.js [hunk 2]
```diff
@@ -469,4 +469,41 @@ Then('the promo image is matching the earlier selected image', () => {
...
+
+And('user clicks on title with url spanish path {string} site section plus {string}', (spPath, purl) => {
+    cy.get(`a[href='${spPath}${siteSection}/${purl}-${randomNum}']`).click();
+});
+
+And('list row card title is {string}', (listTitleSpPath) => {
+    cy.get(`div[class="dynamic list"] h2:contains("${listTitleSpPath}")`).should('be.visible')
+})
+
+And('every link in dynamic list starts with {string}', (spanolPath) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$HomeAndLandingPage/index.js:505]: please change spelling of param -> espanolPath

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

**Comment by ahmadakbaryNIH** [comment]
image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**